src/viper/trans.ml

```diff
@@ -100,6 +100,8 @@ module Env = T.Env
 type ctxt =
   { self : string option;
     ids : (sort * T.t) T.Env.t;
+    label_vars : (string, id) Hashtbl.t; (* Orig label var -> associated tmp variable. *)
+    mapped_labels : (string, string) Hashtbl.t; (* Orig label var -> label var in viper. *)
```

Comment:
If we make the translation of label names use a deterministic scheme (like the one I suggest in https://github.com/serokell/motoko/pull/18/files#r1621415198), then we don't need this hashmap, do we? Just have a toplevel `string -> string` function.

As I understand it, the current translation makes use of `fresh_id`, hence nondeterminism and the need to track those labels in the context. But it's not clear to me what problem we are solving with this use of `fresh_id`. Does it help us to avoid conflicts? If yes, could you demonstrate when such conflicts arise, is there an example?

Reply:
> But it's not clear to me what problem we are solving with this use of fresh_id

Labels in `viper` couldn't be shadowed, so, we need to use `fresh_id` here.

> could you demonstrate when such conflicts arise, is there an example?

Yeah, `label-break-continue.mo` already has such example. Let me point it out:
```
    let block_label_early_expr = label block : (Int, Int) {
      if (true) break block(42, 42);
      (24, 24)
    };

    let block_label_expr = label block : (Int, Int) {
      if (false) break block(42, 42);
      (24, 24)
    };
```
Here we want to distinguish these two `block` labels

Reply:
OK, I see, thank you for the explanation. I also see that it is covered by a test case, good.